packages/pricing/src/interfaces/pricing.interface.ts

```diff
@@ -1,16 +1,18 @@
+import { Address } from "@zkchainhub/shared";
+
 // providers
 export type PricingProvider = "coingecko";
 
 /**
  * Represents a pricing service that retrieves token prices.
+ * @dev is service responisility to map address to their internal ID
+ * @dev for native token (eg. ETH), use the one address
  */
 export interface IPricingProvider {
     /**
      * Retrieves the prices of the specified tokens.
-     * @param tokenIds - An array of token IDs.
-     * @returns A promise that resolves to a record containing the token IDs as keys and their corresponding prices as values.
+     * @param addresses - An array of token addresses.
+     * @returns A promise that resolves to a record containing the token address as keys and their corresponding prices or undefined as values
      */
-    getTokenPrices<TokenId extends string = string>(
-        tokenIds: TokenId[],
-    ): Promise<Record<string, number>>;
+    getTokenPrices(addresses: Address[]): Promise<Record<Address, number | undefined>>;
```

Comment:
I've seen the `Record<Address, number | undefined>` type in some places over this PR; should you give it some more entity by creating a new type based on it or do you prefer to leave it as is?